src/IR.h

```diff
@@ -774,6 +774,14 @@ struct Shuffle : public ExprNode<Shuffle> {
      * arguments. */
     bool is_interleave() const;
 
+    /** Check if this shuffle can be represented as a broadcast of Type t.
+     * For example:
+     * A uint8 shuffle of with 4*n lanes and indices:
+     *     0, 1, 2, 3, 0, 1, 2, 3, ....., 0, 1, 2, 3
+     * can be treated as a uint32 broadcast with n lanes.
+     * Return a Broadcast if possible, Expr() otherwise. */
```

Comment:
This should return true/false, and just accept a broadcast factor. The caller should be responsible for making a broadcast op if it wants one.